google-cloud-clients/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/JobStatistics.java

```diff
@@ -754,18 +761,250 @@ static QueryStatistics fromPb(
+          this.startLine = startLine;
+          return this;
+        }
+
+        Builder setText(String text) {
+          this.text = text;
+          return this;
+        }
+
+        ScriptStackFrame build() {
+          return new ScriptStackFrame(this);
+        }
+      }
+
+      public Integer getEndColumn() {
+        return endColumn;
+      }
+
+      public Integer getEndLine() {
+        return endLine;
+      }
+
+      public String getProcedureId() {
+        return procedureId;
+      }
+
+      public Integer getStartColumn() {
+        return startColumn;
+      }
+
+      public Integer getStartLine() {
+        return startLine;
+      }
+
+      public String getText() {
+        return text;
+      }
+
+      com.google.api.services.bigquery.model.ScriptStackFrame toPb() {
+        com.google.api.services.bigquery.model.ScriptStackFrame stackFrame =
+            new com.google.api.services.bigquery.model.ScriptStackFrame();
+        stackFrame.setEndColumn(endColumn);
+        stackFrame.setEndLine(endLine);
+        stackFrame.setProcedureId(procedureId);
+        stackFrame.setStartColumn(startColumn);
+        stackFrame.setStartLine(startLine);
+        stackFrame.setText(text);
+        return stackFrame;
+      }
+
+      static Builder newBuilder() {
+        return new Builder();
+      }
+
+      static ScriptStackFrame fromPb(
+          com.google.api.services.bigquery.model.ScriptStackFrame stackFrame) {
+        Builder builder = newBuilder();
+        if (stackFrame.getEndColumn() != null) {
+          builder.setEndColumn(stackFrame.getEndColumn());
+        }
+        if (stackFrame.getEndLine() != null) {
+          builder.setEndLine(stackFrame.getEndLine());
+        }
+        if (stackFrame.getProcedureId() != null) {
+          builder.setProcedureId(stackFrame.getProcedureId());
+        }
+        if (stackFrame.getStartColumn() != null) {
+          builder.setStartColumn(stackFrame.getStartColumn());
+        }
+        if (stackFrame.getStartLine() != null) {
+          builder.setStartLine(stackFrame.getStartLine());
+        }
+        if (stackFrame.getText() != null) {
+          builder.setText(stackFrame.getText());
+        }
+        return builder.build();
+      }
+    }
+
+    private ScriptStatistics(Builder builder) {
+      this.evaluationKind = builder.evaluationKind;
+      this.stackFrames = builder.stackFrames;
+    }
+
+    static class Builder {
+
+      private String evaluationKind;
+      private List<ScriptStackFrame> stackFrames;
+
+      private Builder() {}
+
+      Builder setEvaluationKind(String evaluationKind) {
+        this.evaluationKind = evaluationKind;
+        return this;
+      }
+
+      Builder setStackFrames(List<ScriptStackFrame> stackFrames) {
+        this.stackFrames = stackFrames;
+        return this;
+      }
+
+      ScriptStatistics build() {
+        return new ScriptStatistics(this);
+      }
+    }
+
+    static Builder newBuilder() {
```

Comment:
Are end-users suppose to build a `ScriptStatistics` object? 

If yes then we should make this builder public. If not then should we really need to make `ScriptStatistics` a `public` class?

Reply:
`ScriptStatistics` only output field